DeusEx/Classes/DXREntranceRando.uc

```diff
@@ -317,149 +319,70 @@ function GenerateConnections(int missionNum)
-    xfers[10].outTag = "FromRoofTop";
-    xfers[10].used = False;
-
-    xfers[11].mapname = mapname;
-    xfers[11].inTag = "ToWarehouseAlley";
-    xfers[11].outTag = "FromWarehouseAlley";
-    xfers[11].used = False;
-
-    mapName = "02_NYC_Street";
-    xfers[12].mapname = mapname;
-    xfers[12].inTag = "ToStreet";
-    xfers[12].outTag = "ToBatteryPark";
-    xfers[12].used = False;
-
-    xfers[13].mapname = mapname;
-    xfers[13].inTag = "FromSmugBackDoor";
-    xfers[13].outTag = "ToSmugBackDoor";
-    xfers[13].used = False;
-
-    xfers[14].mapname = mapname;
-    xfers[14].inTag = "FromSmugFrontDoor";
-    xfers[14].outTag = "ToSmugFrontDoor";
-    xfers[14].used = False;
-
-    xfers[15].mapname = mapname;
-    xfers[15].inTag = "FromBarBackEntrance";
-    xfers[15].outTag = "ToBarBackEntrance";
-    xfers[15].used = False;
-
-    xfers[16].mapname = mapname;
-    xfers[16].inTag = "FromBarFrontEntrance";
-    xfers[16].outTag = "ToBarFrontEntrance";
-    xfers[16].used = False;
-
-    xfers[17].mapname = mapname;
-    xfers[17].inTag = "FromHotelFrontDoor";
-    xfers[17].outTag = "ToHotelFrontDoor";
-    xfers[17].used = False;
-
-    xfers[18].mapname = mapname;
-    xfers[18].inTag = "FromClinic";
-    xfers[18].outTag = "FromStreet";
-    xfers[18].used = False;
-
-    xfers[19].mapname = mapname;
-    xfers[19].inTag = "BedroomWindow";
-    xfers[19].outTag = "ToHotelBedroom";
-    xfers[19].used = False;
-
-    xfers[20].mapname = mapname;
-    xfers[20].inTag = "FromWarehouseAlley";
-    xfers[20].outTag = "ToWarehouseAlley";
-    xfers[20].used = False;
-
-    xfers[21].mapname = mapname;
-    xfers[21].inTag = "FromRoofTop";
-    xfers[21].outTag = "ToRoofTop";
-    xfers[21].used = False;
-
-    xfers[22].mapname = mapname;
-    xfers[22].inTag = "FromNYCUndergroundSewer2";
-    xfers[22].outTag = "ToNYCUndergroundSewer2";
-    xfers[22].used = False;
-
-    xfers[23].mapname = mapname;
-    xfers[23].inTag = "FromNYCSump";
-    xfers[23].outTag = "ToNYCSump";
-    xfers[23].used = False;
-
+    numXfers++;
+}
 
+function RandoMission2()
+{
+    
+    AddXfer("02_NYC_BatteryPark","ToBatteryPark","ToStreet");
+    AddXfer("02_NYC_Bar","ToBarBackEntrance","FromBarBackEntrance");
+    AddXfer("02_NYC_Bar","ToBarFrontEntrance","FromBarFrontEntrance");
+    AddXfer("02_NYC_FreeClinic","FromStreet","FromClinic");
+    AddXfer("02_NYC_Hotel","ToHotelBedroom","BedroomWindow");
+    AddXfer("02_NYC_Hotel","ToHotelFrontDoor","FromHotelFrontDoor");
+    AddXfer("02_NYC_Smug","ToSmugFrontDoor","FromSmugFrontDoor");
+    AddXfer("02_NYC_Smug","ToSmugBackDoor","FromSmugBackDoor");
+    AddXfer("02_NYC_Underground","ToNYCUndergroundSewer2","FromNYCUndergroundSewer2");
+    AddXfer("02_NYC_Underground","ToNYCSump","FromNYCSump");
+    AddXfer("02_NYC_Warehouse","ToRoofTop","FromRoofTop");
+    AddXfer("02_NYC_Warehouse","ToWarehouseAlley","FromWarehouseAlley");
+    AddXfer("02_NYC_Street","ToStreet","ToBatteryPark");
```

Comment:
isn't this kinda redundant? or maybe error prone repetition?

```
AddXfer("02_NYC_BatteryPark","ToBatteryPark","ToStreet");
//....
AddXfer("02_NYC_Street","ToStreet","ToBatteryPark");
```

I feel like you could just do
`AddXfers("02_NYC_BatteryPark","ToBatteryPark","02_NYC_Street", "ToStreet");`

Reply:
I think it's a bit dangerous to define them as the connection between maps - The idea I was going for was to define the edge points of a map.  Yes, the way you suggest might be a bit more efficient, but I don't think it describes the actual entities that we care about.  Remember, each map edge has two teleporters in each map